### PR TITLE
[M@Sv2] OST: Solution design

### PR DESCRIPTION
**Pull request by rohitsahu**
- Added OST V2 Initialization and Web Component, The purpose of this Draft PR is to drive discussion on the data model b/w OST and Studio framework

Fix #<[MWPW-152810](https://jira.corp.adobe.com/browse/MWPW-152810)>

Test URLs:
- Before: https://main--mas--adobecom.hlx.live/
- After: https://ost--mas--adobecom.hlx.live/

### CODE REVIEW

File: studio/libs/ost.css
Comment (yesil) [inline studio/libs/ost.css:1]: a simpler solution comes for the actual OST.
I guess this is not supposed to be in your PR.

File: studio/libs/ost.js
Comment (yesil) [inline studio/libs/ost.js:1]: you can remove this from your PR

File: studio/libs/studio.js
Comment (yesil) [inline studio/libs/studio.js:1]: you can also omit this from your PR.

File: ost/package.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{
+  "name": "@adobe/ost",
```
[the remark this answers is not in the document]
Comment (Axelcureno) [reply ost/package.json:2]: @adobe/offer-selector-tool sounds explanatory

[the remark this answers is not in the document]
Comment (npeltier) [reply ost/package.json:2]: yep, @adobe/offer-selector-tool

File: ost/src/service/NetworkService.js [hunk 2]
```diff
@@ -0,0 +1,113 @@
+import { computePlanType } from '../utils/Utils.js';
+
+export class NetworkService {
+
+    constructor(accessToken, apiKey) {
+        this.accessToken = accessToken;
+        this.apiKey = apiKey;
+    }
+    
+    loadScript(src) {
```
[the remark this answers is not in the document]
Comment (npeltier) [reply ost/src/service/NetworkService.js:10]: agreed, but let's not do it in that PR

File: ost/src/service/NetworkService.js
[the remark this answers is not in the document]
Comment (npeltier) [reply ost/src/service/NetworkService.js:1]: same, agreed but not int that PR

File: ost/src/service/NetworkService.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (rohitsahu) [reply ost/src/service/NetworkService.js:12]: I hace added a comment

File: ost/src/service/NetworkService.js [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        const KEY = {
+            PA: 'product_arrangement_code',
+        };
+        const [, country] = locale.split('_');
+        const OFFERS_ENDPOINT = `https://aos.adobe.io/offers?country=${country}&merchant=ADOBE&service_providers=MERCHANDISING&locale=${locale}&api_key=YOUR_API_KEY&landscape=${landscape}&page_size=100&page=${page}`;
+        
+        return fetch(OFFERS_ENDPOINT)
+            .then((response) => response.json())
+            .then(async (offers) => {
+                console.log(`received ${landscape} - ${page}`);
```
[the remark this answers is not in the document]
Comment (rohitsahu) [reply ost/src/service/NetworkService.js]: I could not properly understand how the Log handle was available after the service initiatialisation. 
Also, will this logging approach also work if OST function as standalone?

File: mas/src/mas.js [hunk 4]
```diff
@@ -19,5 +19,7 @@ const config = () => ({
 init(config);
 
 if (features.includes('merch-card')) {
-    import(`${origin}/libs/merch-card-all.js`);
+    import(
+        `https://main--milo--adobecom.hlx.page/libs/deps/mas/merch-card-all.js`
```
Comment (Axelcureno) [inline mas/src/mas.js:23]: a change is coming here in a separate PR just FYI.

File: ost/README.md [hunk 5]
```diff
@@ -0,0 +1 @@
+# OST - Offer Selection Tool
```
Comment (Axelcureno) [inline ost/README.md:1]: Offer Selector Tool :)

File: mas/src/mas.js (hunk 4 above)
Comment (rohitsahu) [reply mas/src/mas.js:23]: Got it

File: ost/src/service/NetworkService.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (rohitsahu) [reply ost/src/service/NetworkService.js]: And would it be okay, if I take the logging solution separately?

File: ost/src/service/NetworkService.js [hunk 6]
```diff
@@ -0,0 +1,128 @@
+import { computePlanType } from '../utils/Utils.js';
```
Comment (Axelcureno) [inline ost/src/service/NetworkService.js:1]: This file doesn't exist

**Comment by yesil** [comment]
closing for now.

**Pull request closed without merging**